src/pages/index.js

```diff
@@ -134,5 +224,52 @@ profilePopup.setEventListeners();
 const cardPopup = new PopupWithForm(addModalClassStg, handleAddCardFormSubmit);
 cardPopup.setEventListeners();
 
+/*---------------------------------------------------*/
+/*          PopupWithImage Constructor               */
+/*---------------------------------------------------*/
+
 const popupImg = new PopupWithImage(previewModalClassStg);
-popupImg.setEventListeners();
+
+/*---------------------------------------------------*/
+/*         PopupWithCofirm Constructor               */
+/*---------------------------------------------------*/
+
+const trashConfirmPopup = new PopupWithConfirm(
+  trashModalClassStg,
+  handleDeleteConfirmModal
+);
+trashConfirmPopup.setEventListeners();
+
+/*---------------------------------------------------*/
+/*                      Api                          */
+/*---------------------------------------------------*/
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: authorizationCode,
+  },
+});
+
+const profileApiObject = api.getProfile();
+
+profileApiObject.then((data) => {
+  userInfo.setUserInfo(data.name, data.about);
+  avatarPopup.handleAvatarChange(data.avatar);
+});
+
+/*---------------------------------------------------*/
+/*               Section Constructor                 */
+/*---------------------------------------------------*/
+// To get the size of cardArray outside the getUniqueCards function,
+// you need to handle it asynchronously since getUniqueCards has
+// asynchronous logic inside. You can return a promise from getUniqueCards
+// and then handle it accordingly.
+const uniqueCards = getUniqueCards();
+
+const section = new Section(
+  { items: uniqueCards, renderer: createCard },
+  ".card__list"
+);
+
+section.renderItems();
```

Comment:
You can't guarantee that `uniqueCards` will come first and then `section.renderItems();` will be called because of the server delay.


You should change the logics. You should call `section.renderItems(cards);` in block `then` after the cards has been received

Reply:
Hey @gennady-bars, I am not sure if this is what you are referring to in your latest pull request comment: "You haven't corrected some of my remarks from here . . . " because I have since updated the code referenced above and updated the function 'renderItems()' in 'section.js'.

Is the code you are referring to in you comment in the last pull request? 

Newer Code.
![Screenshot 2024-08-21 143103](https://github.com/user-attachments/assets/f1df7a4f-ea65-4119-bae2-9c2b17035d1c)
![Screenshot 2024-08-21 143126](https://github.com/user-attachments/assets/3db5504e-2dd9-44be-9872-e93692832ff8)
![Screenshot 2024-08-21 143143](https://github.com/user-attachments/assets/67cbd889-96f7-4df2-9cfd-7815f2f1ff14)